Heavy memory usage

# 🐛 bug report/question

Trying to import `ffmpeg.js` results in a process taking up 5GBs+ of RAM in a few seconds.
The imported file is 10mb so i can understand it taking up a lot of ram but that much ?

This is something that has been recurring over my projects, parcel processes taking up a lot of ram.
Do you guys have any plans on reducing the memory footprint ?

Here is the profile:
[profile-20210109-161111.zip](https://github.com/parcel-bundler/parcel/files/5791916/profile-20210109-161111.zip)

![image](https://user-images.githubusercontent.com/16418197/104109677-0c4ee080-5296-11eb-9d5c-7956fa22e775.png)



BTW, the profiler seems to have issues in writing the files to the disk. I have to run the profiler a couple of times (with clean in between) to get one working. It exits with a crash code.
```
@parcel/core: Starting profiling...
/ Optimizing SimpleTable.js...
npm ERR! code 3221225477
```

--- a/packages/core/workers/src/WorkerFarm.js
+++ b/packages/core/workers/src/WorkerFarm.js
@@ -628,7 +628,7 @@ export default class WorkerFarm extends EventEmitter {
   static getNumWorkers(): number {
     return process.env.PARCEL_WORKERS
       ? parseInt(process.env.PARCEL_WORKERS, 10)
-      : Math.ceil(cpuCount() / 2);
+      : Math.min(4, Math.ceil(cpuCount() / 2));
   }
 
   static isWorker(): boolean {

